sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2236,6 +2236,19 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val STATE_STORE_COORDINATOR_MIN_SNAPSHOT_VERSION_DELTA_TO_LOG =
+    buildConf("spark.sql.streaming.stateStore.minSnapshotVersionDeltaToLog")
+      .internal()
+      .doc(
+        "Minimum number of versions between the most recent uploaded snapshot version of a " +
+        "single state store instance and the most recent version across all state store " +
+        "instances to log a warning message."
+      )
+      .version("4.0.0")
+      .intConf
+      .checkValue(k => k >= 0, "Must be greater than or equal to 0")
```

Comment:
what happens if equal to 0? Should we be able to turn this feature off?

Reply:
Good point, if it's 0 it will just report every instance as lagging which doesn't seem ideal. I will add another config to enable/disable these event reports to the coordinator 👍